server/src/test/java/io/pravega/schemaregistry/storage/impl/group/GroupTest.java

```diff
@@ -0,0 +1,592 @@
+import io.pravega.schemaregistry.contract.data.SchemaInfo;
+import io.pravega.schemaregistry.contract.data.SchemaWithVersion;
+import io.pravega.schemaregistry.contract.data.SerializationFormat;
+import io.pravega.schemaregistry.contract.data.VersionInfo;
+import io.pravega.schemaregistry.storage.Etag;
+import io.pravega.schemaregistry.storage.StoreExceptions;
+import io.pravega.schemaregistry.storage.impl.group.records.TableRecords;
+import io.pravega.test.common.AssertExtensions;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import java.nio.ByteBuffer;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.Executors;
+import java.util.stream.Collectors;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+public class GroupTest {
+    String groupId;
+    InMemoryGroupTable integerGroupTable;
+    Group<Integer> integerGroup;
+    private ScheduledExecutorService executor;
+
+    @Before
+    public void setup() {
+        executor = Executors.newScheduledThreadPool(5);
+        groupId = "mygroup";
+        integerGroupTable = new InMemoryGroupTable();
+        integerGroup = new Group<Integer>(integerGroupTable, executor);
+    }
+
+    @After
+    public void tearDown() {
+        executor.shutdownNow();
+    }
+
+    @Test
+    public void testCreate() {
+        integerGroup.create(SerializationFormat.Custom, ImmutableMap.of(), Boolean.TRUE,
+                Compatibility.backward()).join();
+        List<TableRecords.TableValue> recordList = integerGroupTable.table.entrySet().stream().map(
+                x -> x.getValue().getValue()).collect(Collectors.toList());
+        assertEquals(Boolean.TRUE, recordList.contains(
+                new TableRecords.ValidationRecord(Compatibility.backward())));
+    }
+
+    @Test
+    public void testGetCurrentEtag() {
+        // null case
+        Etag eTag = integerGroup.getCurrentEtag().join();
+        assertEquals(null, integerGroupTable.fromEtag(eTag));
+        assertTrue(integerGroupTable.table.isEmpty());
+        // non null case
+        integerGroup.create(SerializationFormat.Custom, ImmutableMap.of(), Boolean.TRUE,
+                Compatibility.backward()).join();
+        List<Integer> integerList = integerGroupTable.table.entrySet().stream().filter(
+                x -> x.getValue().getValue().equals(
+                        new TableRecords.ValidationRecord(Compatibility.backward()))).map(
+                x -> x.getValue().getVersion()).collect(
+                Collectors.toList());
+        assertEquals(1, integerList.size());
+        eTag = integerGroup.getCurrentEtag().join();
+        assertEquals(integerList.get(0), integerGroupTable.fromEtag(eTag));
+    }
+
+    @Test
+    public void testgetTypes() {
```

Comment:
here and elsewhere, please follow camel casing in naming of methods

Reply:
done